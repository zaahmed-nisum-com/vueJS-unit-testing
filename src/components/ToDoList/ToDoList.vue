<template>
  <div>
    <div v-for="(item, index) of items" v-bind:key="index + 1">
      <ToDoItem :item="item" />
    </div>
  </div>
</template>

<script>
import ToDoItem from "../ToDoItem/ToDoItem.vue";
export default {
  components: { ToDoItem },
  props: ["items"],
};
</script>

<style></style>
