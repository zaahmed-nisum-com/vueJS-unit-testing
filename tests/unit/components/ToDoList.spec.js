import { createLocalVue, mount, shallowMount } from '@vue/test-utils'
import ToDoList from '../../../src/components/ToDoList/ToDoList.vue'


describe('ToDoList.vue', () => {
    test('snapshot', () => {
        const wrapper = mount(ToDoList, {
            propsData: {
                item: []
            }
        })
        expect(wrapper.find('div')).toMatchSnapshot(true)
    })

    test('check array props', () => {
        const wrapper = mount(ToDoList, {
            propsData: {
                item: {}
            }
        })
        expect(wrapper.props().item).toEqual(expect.arrayContaining([]))
    })

})
