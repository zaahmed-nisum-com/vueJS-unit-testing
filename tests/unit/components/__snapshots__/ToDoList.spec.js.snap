// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue snapshot 1`] = `<div></div>`;

exports[`ToDoList.vue snapshot 1`] = `<div item=""></div>`;
